src/gen/gen_ast_builder.ml

```diff
@@ -140,6 +248,26 @@ let dump fn ~ext printer x =
   Format.fprintf ppf "%a@." printer x;
   close_out oc
 
+let floating_comment s =
+  let doc =
+    PStr
+      [
+        {
+          pstr_desc =
+            Pstr_eval
+              ( {
+                  pexp_desc = Pexp_constant (Pconst_string (s, loc, None));
+                  pexp_loc = loc;
+                  pexp_loc_stack = [];
+                  pexp_attributes = [];
+                },
+                [] );
+          pstr_loc = loc;
+        };
+      ]
+  in
+  Sig.attribute (Attr.mk { txt = "ocaml.text"; loc } doc)
```

Comment:
Can't we simplify this with `M.sigi`?

Reply:
The parser seemed to feel on OCaml 4.04 and 4.05 when parsing a floating documentation comment no matter what way I expressed it (`(** {2 ...} *)`) or with the attribute

Reply:
Ah ok, good to know! Let's keep it that way then!